File: WebClient/src/app/model/irrigationaction.ts
export class IIrrigationAction {
    constructor(
        public id: number,
        public Name: string,
        public Start: Date,
        public Finished: Date,
        public Paused: Date,
        public Duration: number,
        public Progress: number,
        public SolenoidId: number,
        public RequiresPump: boolean,
        public DeviceId: number,
        public createdAt: Date,
        public updatedAt: Date
    ) {}
}

File: Device/DeviceController.Data/IrrigationAction.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DeviceController.Data
{
    public class IrrigationAction
    {
        public int Id;
        public string Name;
        public DateTime Start;
        public DateTime? Finished;
        public DateTime? Paused;
        public int Duration;
        public int Progress;
        public int DeviceId;
        public int SolenoidId;
        public string SolenoidName;        
        public bool RequiresPump;        
    }
}
